packages/components/src/internal/components/samples/SampleAliquotsGridPanel.spec.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+
+import {
+    App,
+    LoadingSpinner,
+    LoadingState,
+    QueryInfo,
+    QueryModel,
+    SampleAliquotsGridPanel,
+    SchemaQuery,
+} from '../../../index';
+import { mountWithServerContext } from '../../testHelpers';
+
+describe('<SampleAliquotsGridPanel/>', () => {
+    const SAMPLE_TYPE_NAME = 'SampleTypeName';
+    const DEFAULT_CONTEXT = { user: App.TEST_USER_EDITOR };
+
+    const queryInfo = QueryInfo.create({ schemaName: 'samples', name: SAMPLE_TYPE_NAME, queryLabel: SAMPLE_TYPE_NAME });
+    let model = new QueryModel({ schemaQuery: SchemaQuery.create('samples', SAMPLE_TYPE_NAME) });
+    model = model.mutate({
+        queryInfo,
+        queryInfoLoadingState: LoadingState.LOADED,
+        rowsLoadingState: LoadingState.LOADED,
+        chartsLoadingState: LoadingState.LOADED,
+    });
+
+    const DEFAULT_PROPS = {
+        sampleLsid: 'lsidValue',
+        schemaQuery: SchemaQuery.create('samples', SAMPLE_TYPE_NAME),
+        user: App.TEST_USER_READER,
+        onSampleChangeInvalidate: jest.fn(),
+        queryModels: { modelid: model },
+        actions: {
```

Comment:
These `actions` and `queryModels` are actually instantiated by `withQueryModels` so passing them into `SampleAliquotsGridPanel` does nothing. Instead, I would recommend exporting `SampleAliquotsGridPanelImpl` and test against that directly.

Additionally, you can make use of `makeTestActions` to initialize the actions.

```ts
actions: makeTestActions(jest.fn),
```

Reply:
Done (in patch)